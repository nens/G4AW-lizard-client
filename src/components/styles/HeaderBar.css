.HeaderBar {
  width: 100%;
  height: 50px;
  background-color: #d8d8d8;
  color: #fff;
  text-transform: uppercase;
  line-height: 50px;
  font-size: 18px;
  padding: 0 0 0 0;
  margin: 10px 0 0 0;
  position: relative;
}

.ActionButton {
  position: absolute;
  cursor: pointer;
  right: 25px;
  top: 5px;
}

.Title {
  left: 15px;
  position: absolute;
}
