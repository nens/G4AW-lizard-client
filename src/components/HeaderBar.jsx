import React, { Component } from "react";
import PropTypes from "prop-types";
import ReactDOM from "react-dom";
import styles from "./styles/HeaderBar";

// A HeaderBar component.

export default class HeaderBar extends Component {
  render() {
    const { title, icon, handleClick } = this.props;
    return (
      <div className={styles.HeaderBar} id="HeaderBar">
        <span className={styles.Title}>{title}</span>
        <div className={styles.ActionButton} onClick={handleClick}>
          <i className={`${styles.ActionButtonIcon} material-icons`}>{icon}</i>
        </div>
      </div>
    );
  }
}

HeaderBar.propTypes = {
  handleClick: PropTypes.func,
  icon: PropTypes.string,
  title: PropTypes.string
};
