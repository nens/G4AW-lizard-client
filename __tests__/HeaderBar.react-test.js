// HeaderBar.react-test.js
import React from "react";
import HeaderBar from "../src/components/HeaderBar";
import renderer from "react-test-renderer";

test("Component shows word Test as the title", () => {
  const component = renderer.create(<HeaderBar title="Test" />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
