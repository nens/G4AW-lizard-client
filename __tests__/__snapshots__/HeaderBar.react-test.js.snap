// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component shows word Test as the title 1`] = `
<div
  className="HeaderBar"
>
  Test
  <div
    className="ActionButton"
    onClick={undefined}
  >
    <i
      className="ActionButtonIcon material-icons"
    />
  </div>
</div>
`;
